tests/publisher_test.py

```diff
@@ -404,3 +404,112 @@ def testUsesUnionOfMetaKeys(self):
+          "metadata={'info': '1','bar': 'foo', 'more_info': '2'}", 'value=non',
+          'sample_uri=33']
+
+    self.assertEqual(self.sortList(sorted(no_meta_keys)),
+                     self.sortList(sorted(s1)))
+    self.assertEqual(self.sortList(sorted(empty_meta_keys)),
+                     self.sortList(sorted(s2)))
+    self.assertEqual(self.sortList(sorted(with_meta_keys)),
+                     self.sortList(sorted(s3)))
+
+  def testConstructSample(self):
+    no_meta_samples = self.test_db._ConstructSample(self.no_meta)
+    empty_meta_samples = self.test_db._ConstructSample(self.empty_meta)
+    with_meta_samples = self.test_db._ConstructSample(self.with_meta)
+
+    l1 = ('perfkitbenchmarker,run_uri=5rtw,test=testa,sample_uri=5r,metric=3,'
+          'official=47.0,owner=Rackspace,unit=us value=non 123000000000')
+    l2 = ('perfkitbenchmarker,metric=2,official=14.0,owner=Rackspace,'
+          'run_uri=bba3,test=testb,unit=MB,sample_uri=bb '
+          'value=non 55000000000')
+    l3 = ('perfkitbenchmarker,metric=1,official=1.0,owner=Rackspace,'
+          'run_uri=323,test=testc,unit=MB,sample_uri=33,info=1,'
+          'bar=foo,more_info=2 value=non 123000000000')
+
+    self.assertEqual(self.sortString(no_meta_samples), self.sortString(l1))
+    self.assertEqual(self.sortString(empty_meta_samples), self.sortString(l2))
+    self.assertEqual(self.sortString(with_meta_samples), self.sortString(l3))
+
+  def testAggregationOfSamples(self):
+    sample_data = [{'test': 'testc', 'metric': '1', 'official': 1.0,
+                   'value': 'non', 'unit': 'MB', 'owner': 'Rackspace',
+                    'run_uri': '323', 'sample_uri': '33', 'timestamp': 123,
+                    'metadata':
+                    {'info': '1', 'more_info': '2', 'bar': 'foo'}},
+                   {'test': 'testb', 'metric': '2', 'official': 14.0,
+                    'value': 'non', 'unit': 'MB', 'owner': 'Rackspace',
+                    'run_uri': 'bba3', 'sample_uri': 'bb', 'timestamp': 55,
+                    'metadata': {}},
+                   {'test': 'testa', 'metric': '3', 'official': 47.0,
+                   'value': 'non', 'unit': 'us', 'owner': 'Rackspace',
+                    'run_uri': '5rtw', 'sample_uri': '5r', 'timestamp': 123}]
+    formated_samples = []
+    s1 = ('perfkitbenchmarker,sample_uri=33,owner=Rackspace,official=1.0,'
+          'run_uri=323,metric=1,test=testc,unit=MB,bar=foo,more_info=2,'
+          'info=1 value=non 123000000000 \n perfkitbenchmarker,'
+          'sample_uri=bb,owner=Rackspace,official=14.0,run_uri=bba3,metric=2,'
+          'test=testb,unit=MB value=non 55000000000 \n perfkitbenchmarker,'
+          'sample_uri=5r,owner=Rackspace,official=47.0,run_uri=5rtw,metric=3,'
+          'test=testa,unit=us value=non 123000000000')
+    expected_sample_format = self.sortString(s1)
+    for data in sample_data:
+      formated_samples.append(self.test_db._ConstructSample(data))
```

Comment:
Since `_ConstructSample(data)` returns a string, why not simply test for that? No need to join them or append them in a list.

If you want to test the expected data format that gets sent to Influx, I'd suggest then to stub out `_CreateDB()` and `_WriteData()` and add a test for `_Publish()` to assert that `_WriteData` gets called with the data format you expect.